docker/nginx/conf.d/client.conf

```diff
@@ -127,10 +128,42 @@ server {
 	}
 
 	location /hns {
-		include /etc/nginx/conf.d/include/cors;
 		include /etc/nginx/conf.d/include/proxy-buffer;
 
-		proxy_pass http://handshake-api:3100;
+		set $skylink ''; # placeholder for the raw 46 bit skylink
+		set $rest ''; # placeholder for the rest of the url that gets appended to skylink (path and args)
+		
+		# resolve handshake domain by requesting to /hnsres endpoint and assign correct values to $skylink and $rest
+		access_by_lua_block {
+			local json = require('cjson')
+			local hns_domain_name, request_uri_rest = string.match(ngx.var.request_uri, "/hns/([^/?]+)(.*)")
+			local hnsres_res = ngx.location.capture("/hnsres/" .. hns_domain_name)
+			local hnsres_json = json.decode(hnsres_res.body)
+			local skylink_prefix, skylink, skylink_rest = string.match(hnsres_json.skylink, "(sia://)([^/?]+)(.*)")
+
+			ngx.var.skylink = skylink
+			if request_uri_rest == "" or (request_uri_rest == "/" and skylink_rest ~= "") then
+				ngx.var.rest = skylink_rest
+			else
+				ngx.var.rest = request_uri_rest
+			end
+		}
+
+		# overwrite the Cache-Control header to only cache for 60s in case the domain gets updated
```

Comment:
What does this mean exactly after we've proxy passed it to localhost again which should handle caching right? Isn't the setting of the header redundant at this point?

Reply:
this is for browser caching - it only sets the header that is consumed by browser

the cache you are referring to is proxy cache inside of nginx and it's not affected by that line